pkg/repository/repository.go

```diff
@@ -75,11 +73,6 @@ func newGitURL(path string) repositoryURL {
 // NewRepository creates a repository controller
 func NewRepository(path string) Repository {
 	var controller repositoryInterface = newGitRepoController(path)
-	// check if we are inside a remote deploy
-	if v := os.Getenv(constants.OktetoDeployRemote); v != "" {
```

Comment:
could you please help me understand why we can also remove this? I assume other cmds go through this logic in remote 🤔

Reply:
Because if not, the images built within the remote won't take advantage of Smart Builds, as functions like `GetDiffHash` or `GetLatestDirSHA` returns a `not-implemented` error.

Most of these checks (checking if we are in remote) were added to prevent duplicated behaviors (and sometimes, getting different results) when `okteto deploy --remote` was executing a `okteto deploy` in BuildKit. Instead of splitting the code into 2 different commands, one the deploy triggered and another one with the logic to only execute the needed logic in remote as we have now, we just "blocked" some execution of logic adding those checks. Some examples are: 
* Check in `build.go` was done to prevent that any image could be built as part of the `okteto deploy` executed in BuildKit when they were already built as part of the local execution of `okteto deploy --remote`
* Remote repository controller was done to prevent any "wrong" calculation as part of the `okteto depoy` executed in BuildKit that could lead to indicate wrong images in `OKTETO_BUILD_XXXX` env vars
* `deployInsideDeployConfigMapHandler` was done to prevent that the `okteto deploy` executed in BuildKit modified the same configmap that was being handled by the local `okteto deploy --remote`.

And probably there are more. 

We need to review carefully where this is being used and review possible side effects, but any `okteto` command executed as part of the list of custom commands of an Okteto manifest should have to calculate properly the repository, as it doesn't have be always the one that was triggered with `okteto deploy --remote`. I could clone a repository as part of the commands and execute an okteto command for that repository